### PR TITLE
Masterbar: Apply Calypso 'Add new site' styles to wp-admin

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes 897-gh-Automattic/dotcom-forge

## Proposed changes

Applies Calypso's 'Add new site' button styles to wp-admin:

image

Note: The button doesn't sit all of the way on the bottom (like it does in Calypso) because `#adminmenuwrap` uses `position: fixed;` to make the sidebar sticky. Given core's admin CSS is a total mess, I don't think it's worth going beyond what I've proposed in this PR.

## Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

## Jetpack product discussion

N/A.

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

1. Create a new free WordPress.com site with a new WordPress.com user.
2. Apply the patch to your sandbox.
3. Sandbox the new WordPress.com site.
4. Navigate to `<sitename>.wordpress.com/wp-admin/` and observe the 'Add new site' styling in a few browsers.
5. Add some free credits to the WordPress.com user.
6. Upgrade the WordPress.com site to a Business plan.
7. Activate hosting features for the WordPress.com site.
8. Install [Jetpack beta plugin](https://jetpack.com/download-jetpack-beta/) and switch to this branch: `update/masterbar-add-new-site-styles`
9. Navigate to `<sitename>.wpcomstaging.com/wp-admin/` and observe the 'Add new site' styling in a few browsers.

### CODE REVIEW

**Review by vykes-mac — DISMISSED** [review]
Works well on Chrome, Firefox and Safari

**Comment by wojtekn** [comment]
@danielbachhuber looks good, it works well for me as well. I've only noticed a small CSS issue in the mobile view - the button in wp-admin doesn't have a space on the right side:

Screen Shot 2022-09-20 at 10 07 37

I played with dev tools and checked what causes it, and it looks like the `li` element gets a `width: 100%` that is set for `.auto-fold #adminmenu li.menu-top` in core WP styles. Setting that to `width: auto;`  should fix the issue.

**Comment by danielbachhuber** [comment]
@vykes-mac @wojtekn Thank you!

> I've only noticed a small CSS issue in the mobile view - the button in wp-admin doesn't have a space on the right side:
> 
> I played with dev tools and checked what causes it, and it looks like the `li` element gets a `width: 100%` that is set for `.auto-fold #adminmenu li.menu-top` in core WP styles. Setting that to `width: auto;` should fix the issue.

@wojtekn Good catch. I've applied and confirmed your fix.

**Review by wojtekn — APPROVED** [review]
Looks good now!

**Comment by kraftbj** [comment]
> core's admin CSS is a total mess
Understatement :)

Giving this a quick look now.

File: projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php [hunk 1]
```diff
@@ -186,7 +186,6 @@ public function add_new_site_link() {
 			return;
 		}
 
-		$this->add_admin_menu_separator();
```
Comment (kraftbj) [inline projects/plugins/jetpack/modules/masterbar/admin-menu/class-atomic-admin-menu.php:189]: Noting that this is still used elsewhere.

**Review by kraftbj — APPROVED** [review]

**Comment by danielbachhuber** [comment]
r252969-wpcom

**Pull request merged**